lib/src/chip/types.ts

```diff
@@ -8,31 +8,60 @@ type Margin = {
 
 type SVG = React.ReactNode & React.SVGProps<SVGSVGElement>;
 
+type Icon = string | SVG;
+
+type PrefixIconProps =
+  | {
+      /**
+       * Element or path used as icon to be placed before the chip label.
+       */
+      prefixIcon: Icon;
+      /**
+       * This function will be called when the prefix is clicked.
+       */
+      onClickPrefix: () => void;
+      /**
+       * Element or path used as icon to be placed after the chip label.
+       */
+      suffixIcon?: Icon;
+      /**
+       * This function will be called when the suffix is clicked.
+       */
+      onClickSuffix?: never;
+      /**
+       * If true, the action icon will be disabled.
+       */
+      disabled?: boolean;
+    }
+  | {
+      prefixIcon?: Icon;
+      onClickPrefix?: never;
+      suffixIcon?: never;
+      onClickSuffix?: () => void;
+      disabled?: never;
+    };

```

Comment:
I think that the typings are not right here. You are allowing the user to have undefined suffixIcon but an onClickSuffix function, which does not make sense.